frame/merkle-mountain-range/src/lib.rs

```diff
@@ -165,14 +175,14 @@ pub mod pallet {
 		///
 		/// Note that the leaf at each block MUST be unique. You may want to include a block hash or
 		/// block number as an easiest way to ensure that.
-		/// Also note that the leaf added by each block is expected to only reference data coming
-		/// from ancestor blocks (leaves are saved offchain using `(parent_hash, pos)` key to be
-		/// fork-resistant, as such conflicts could only happen on 1-block deep forks, which means
-		/// two forks with identical line of ancestors compete to write the same offchain key, but
-		/// that's fine as long as leaves only contain data coming from ancestors - conflicting
-		/// writes are identical).
 		type LeafData: primitives::LeafDataProvider;
 
+		/// The maximum leaf size.
+		type MaxLeafSize: Get<u32>;
+
+		/// The maximum MMR height - log2(max_leaves_count).
+		type MaxMmrHeight: Get<u32>;
```

Comment:
`MaxMmrHeight` is derived from the type used for the `block number` ? (eg: u32: 32, u64: 64) ?

If so maybe we don't `MaxMmrHeight`. Maybe we can have a trait (e.g. `MmrConsts`) with a const value inside (e.g. `MAX_HEIGHT`), implement it for `u32`, `u64`, maybe `u128` and use `<T::BlockNumber as MmrConsts>::MAX_HEIGHT`.

Reply:
That's a good point, but I will actually hardcode it to 32 (meaning u32::Max blocks in MMR). We can come back to it much later if this pallet ever lives to see such a long chain (800+ years on 6sec block chain) 🙈